src/main/scala/com/ubirch/user/client/UserServiceClient.scala

```diff
@@ -418,6 +419,39 @@ object UserServiceClient extends MyJsonProtocol with StrictLogging {
 
   }
 
+  def getUsersByIds(userIds: List[UUID])(implicit httpClient: HttpExt, materializer: Materializer): Future[List[User]] = {
+    logger.debug(s"getUsersByIds(): get users by Ids through REST API")
+
+    Json4sUtil.any2String(userIds) match {
+      case Some(userIdsJson: String) =>
+        val url = UserServiceClientRouteKeys.pathUsersByIds
+        val req = HttpRequest(
+          method = HttpMethods.POST,
+          uri = url,
+          entity = HttpEntity.Strict(ContentTypes.`application/json`, data = ByteString(userIdsJson))
+        )
+        httpClient.singleRequest(req) flatMap {
+          case HttpResponse(StatusCodes.OK, _, entity, _) =>
+
+          entity.dataBytes.runFold(ByteString(""))(_ ++ _) flatMap { body =>
+            Future.fromTry(
+              Try(read[List[User]](body.utf8String))
+            )
+          }
+
+          case _@HttpResponse(code, _, _, _) =>
+
+          logErrorAndReturnNone(s"getUsersWithPagination() call to user-service REST API failed: url=$url, code=$code")
```

Comment:
log has a copy paste issue.
I wonder in general if there should be some limit for the number of userIds somewhere. e.g. on UserService route or here in the client? But if the error handling is good, it's not needed necessarily, I think.

Reply:
> I wonder in general if there should be some limit for the number of userIds somewhere. e.g. on UserService route or here in the client? But if the error handling is good, it's not needed necessarily, I think.

Good point. the user service has to have this responsibility. I will add it to the user service